lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_object_lookup.rb

```diff
@@ -0,0 +1,60 @@
+module MiqAeEngine
+  module MiqAeObjectLookup
+    def initialize_obj_entries
+      @lookup_table = []
+    end
+
+    def add_obj_entry(fqns, klass, instance, object)
+      parts = fqns.split('/')
+      domain = parts.shift
+      namespace = parts.join('/')
+
+      @lookup_table << {:domain    => domain.downcase,
+                        :namespace => namespace.downcase,
+                        :klass     => klass.downcase,
+                        :instance  => instance.downcase,
+                        :object    => object}
+    end
+
+    def find_obj_entry(path)
+      query = path[0] == '/' ? parse_obj_path(path[1..-1]) : parse_obj_path(path)
+
+      entries = @lookup_table.select { |entry| entry[:klass] == query['class_name'] }
+      ientries = entries.select { |entry| instance_match?(query['instance'], entry) }
+      find_best_match(ientries, query)
+    rescue MiqAeException::InvalidPathFormat => err
+      $miq_ae_logger.error(err.message)
+      nil
+    end
+
+    def find_best_match(ientries, query)
+      match = ientries.detect { |entry| fq_match?(query['domain'], query['partial_ns'], entry) }
+      match ||= ientries.detect { |entry| ns_match?(query['fqns'], entry) }
+      match[:object] if match
```

Comment:
@gmcculloug 
Return nil if we can't find a match. The detect gets the first match.